platform/ui/src/utils/ModalProvider.js

```diff
@@ -0,0 +1,91 @@
+import React, { useState, createContext, useContext } from 'react';
+import Modal from 'react-bootstrap-modal';
+
+const ModalContext = createContext(null);
+const { Provider, Consumer } = ModalContext;
+
+export const useModalContext = () => useContext(ModalContext);
+
+const ModalProvider = ({ children }) => {
+  const DEFAULT_OPTIONS = {
+    component: null,
+    backdrop: false,
+    keyboard: false,
+    show: true,
+    large: true,
+    closeButton: true,
+    title: null,
+    customClassName: null,
+  };
+
+  const [options, setOptions] = useState(DEFAULT_OPTIONS);
+
+  const show = (component, props = {}) =>
+    setOptions({
+      ...options,
+      component,
+      ...props,
+    });
+
+  const hide = () => setOptions(DEFAULT_OPTIONS);
```

Comment:
This is neat. So `hide` also "clears" the component.

Reply:
Sure! done, thx.